libraries/standard/mqtt/utest/mqtt_utest.c

```diff
@@ -691,6 +692,17 @@ void test_MQTT_Connect_receiveConnack( void )
     MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTBadResponse );
     status = MQTT_Connect( &mqttContext, &connectInfo, NULL, timeout, &sessionPresent );
     TEST_ASSERT_EQUAL_INT( MQTTBadResponse, status );
+
+    /* Session present is received when a clean session is requested. */
+    mqttContext.transportInterface.recv = transportRecvSuccess;
+    connectInfo.cleanSession = true;
+    sessionPresent = true;
+    MQTT_GetIncomingPacketTypeAndLength_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_GetIncomingPacketTypeAndLength_ReturnThruPtr_pIncomingPacket( &incomingPacket );
+    MQTT_DeserializeAck_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTT_DeserializeAck_ReturnThruPtr_pSessionPresent( &sessionPresent );
```

Comment:
I'd add a variable so the pointer that gets returned by MQTT_DeserializeAck isn't identical to the one passed to MQTT_Connect